Dynamic use of nagios-service checks

laf opened the pull request:
Fix #1792 
Fix #1037

This is a stab at providing this functionality. I added a service check for check_dig which doesn't exist as a current service check for us and this works fine.

It does make two assumptions:

1. The file is called check_*
2. That -H is required and passes the IP / hostname over.

Tidied up the table showing the checks and removed the broken graphs.

Also added the missing data for dbInsert to fix mysql strict bombing out.

bmnnt commented:
ping

f0o commented:
waiting for CI to come back

laf commented:
Bump

f0o commented:
Trying to do a simple ping check returns:
```
<wrta> was not set
check_ping: Could not parse arguments
Usage:
check_ping -H <host_address> -w <wrta>,<wpl>% -c <crta>,<cpl>%
 [-p packets] [-t timeout] [-4|-6]
```

Perhaps we should provide warning/critical parameters as dummies for now or write big notice-banners that the user needs to contribute them via the params form

f0o commented:
Also small Change-Request:
- Hide the details-page button/link
- Order the checks alphabetically

I like the approach, would merge if not for the warning/critical params

f0o commented on html/pages/addsrv.inc.php:
```diff
@@ -18,10 +18,11 @@
-            if ($file != '.' && $file != '..' && !strstr($file, '.')) {
-                $servicesform .= "<option value='$file'>$file</option>";
+            if ($file != '.' && $file != '..' && !strstr($file, '.') && strstr($file, 'check_')) {
+                list(,$check_name) = explode('_',$file);
```
Fix `list(,$check_name) = explode('_',$file);` to `list(,$check_name) = explode('_',$file,2);`. Avoids cutoff where the plugin is called `check_rta_multi` etc.

laf commented:
Added a small note about parameters may be needed.

Added ,2 to the explode.

Checks are already sorted by hostname.

Hide the details link.
